kubernetes/resource_kubernetes_deployment_test.go

```diff
@@ -14,13 +14,38 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-const (
-	defaultNginxImage = "nginx:1.19"
-)
+func TestAccKubernetesDeployment_minimal(t *testing.T) {
+	var conf appsv1.Deployment
+	name := fmt.Sprintf("tf-acc-test-%s", acctest.RandStringFromCharSet(10, acctest.CharSetAlphaNum))
+	imageName := nginxImageVersion
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		IDRefreshName:     "kubernetes_deployment.test",
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckKubernetesDeploymentDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccKubernetesDeploymentConfig_minimal(name, imageName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesDeploymentExists("kubernetes_deployment.test", &conf),
+					resource.TestCheckResourceAttrSet("kubernetes_deployment.test", "metadata.0.generation"),
+					resource.TestCheckResourceAttrSet("kubernetes_deployment.test", "metadata.0.resource_version"),
+					resource.TestCheckResourceAttrSet("kubernetes_deployment.test", "metadata.0.uid"),
+				),
+			},
+			{
+				Config:   testAccKubernetesDeploymentConfig_minimal(name, imageName),
+				PlanOnly: true,
+			},
+		},
+	})
+}
```

Comment:
I added a few tests with the name `_minimal`. The purpose of the minimal tests is to see what happens when the user specifies the bare minimum required to recreate the resource. This tests all of our default values for the resource, and ensures the plan is empty. Non-empty plans have been a problem across a few resources when defaults were applied incorrectly. But they should be fixed now.